### PR TITLE
Throw IMDbIDNotFoundError if we can't figure out the IMDb ID

### PR DESCRIPTION
**Pull request by SubJunk**
I'm interested in your thoughts about this @js-kyle because it exposes a particular inner working of the server to the client which might not be something we want to do. The other option is to throw our existing `MediaNotFoundError`

### CODE REVIEW

**Comment by js-kyle** [comment]
So the custom error is for the API's purposes?

File: src/helpers/customErrors.ts [hunk 1]
```diff
@@ -22,3 +22,12 @@ export class MediaNotFoundError extends Error {
     this.name = 'MediaNotFoundError';
   }
 }
+
+export class IMDbIDNotFoundError extends MediaNotFoundError {
```
Comment (js-kyle) [inline src/helpers/customErrors.ts:26]: I think it would be better to extend the error class?

File: src/app.ts [hunk 2]
```diff
@@ -7,7 +7,7 @@ const debug = Debug('universalmediaserver-api:server');
 
 import indexRouter from './routes/index';
 import mediaRouter  from './routes/media';
-import { ExternalAPIError, MediaNotFoundError } from './helpers/customErrors';
+import { ExternalAPIError, MediaNotFoundError, IMDbIDNotFoundError } from './helpers/customErrors';
```
Comment (js-kyle) [inline src/app.ts:10]: I think we either stick with the generic `MediaNotFoundError` or, add both an imdb one as well as an open subs one

**Comment by SubJunk** [comment]
Both kinda, the error itself is for the API but the new message is for the client, but maybe we shouldn't customise messages for the client like that, and just report either success or failure

File: src/app.ts (hunk 2 above)
Comment (SubJunk) [reply src/app.ts:10]: The error means we couldn't identify what the IMDb ID was based on the title search, so it's not specific to one external API

File: src/helpers/customErrors.ts (hunk 1 above)
Comment (SubJunk) [reply src/helpers/customErrors.ts:26]: It makes more sense now, can you look again?

**Comment by SubJunk** [comment]
@js-kyle conflicts resolved and tests passing

**Pull request merged**